src/array.rs

```diff
@@ -1,98 +1,19 @@
 use crate::{Error, Increment, Key, Metadata, Miniconf};
-use core::ops::{Deref, DerefMut};
 
-/// An array that exposes each element through their [`Miniconf`] implementation.
+/// An array that exposes each element through [`Miniconf`].
 ///
 /// # Design
 ///
-/// Miniconf supports homogeneous arrays of items contained in structures using two forms. For the
-/// [`miniconf::Array`](Array), each item of the array is accessed as a [`Miniconf`] tree.
+/// With `#[miniconf(defer(D))]` and a depth `D > 1` for an
+/// [`[T; N]`](array), each item of the array is accessed as a [`Miniconf`] tree.
+/// For a depth `D = 0`, the entire array is accessed as one atomic
+/// value. By For `D = 1` each index of the array is is instead accessed as
+/// one atomic value.
 ///
-/// For standard arrays of [`[T; N]`](array) form, by default the entire array is accessed as one atomic
-/// value. By adding the `#[miniconf(defer)]` attribute, each index of the array is is instead accessed as
-/// one atomic value (i.e. a single Miniconf item).
-///
-/// The type you should use depends on what data is contained in your array. If your array contains
-/// `Miniconf` items, you can (and often want to) use [`Array`] and the `#[miniconf(defer)]` attribute.
+/// The type to use depends on what data is contained in your array. If your array contains
+/// `Miniconf` items, you can (and often want to) use `D >= 2`.
```

Comment:
It's not really clear to me what a `D = 3` would even do? I guess that's further deferrals through options etc? Maybe it's worth noting that trivial `Miniconf` items such as structs just ignore excessive deferral.

Reply:
e.g. these:

```
#[miniconf(defer(3))]
a: Option<[[i32; 2]; 2]>,
#[miniconf(defer(4))]
a: Option<[Option<[i32; 2]>; 2]>,
```

Reply:
They don't ignore excessive deferral. That'll give you a trait error!

Reply:
A struct (that has Miniconf derived) can only be used with `#[miniconf(defer(1))]` or no miniconf at all. That's why it's lower bound.

Reply:
The awkward interpretation of the trait parameter is resolved by #170 . It's now always just the maximum depth.